src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardController.js

```diff
@@ -32,51 +32,69 @@
         model.getTemplateMetadata().setMode('edit');
     },
 
-    handleButtonClick: function(component, event, helper) {
-        // opt 1: button came from our own component
-        var buttonClick = event.getSource().getLocalId();
-        // opt 2: button came from a ltng:sendMessage
-        // must be explicit about channel because other messages may be sent
-        var channel = event.getParam('channel');
+    /**
+     * @description handles User Input from ltng:sendMessage and onclick handlers
+     * must be explicit about channel because other messages may be sent
+     */
+    handleUserInput: function(component, event, helper) {
+        // check if user input came from ltng:sendMessage or an onclick handler
+        var task;
+        if (event.getSource().getLocalId()) {
+            task = event.getSource().getLocalId();
+        } else if (event.getParam('channel')) {
+            task = event.getParam('channel');
+        }
+
         var model = component.get('v.model');
 
-        if (channel === 'next' || buttonClick === 'next') {
-            var step = component.get('v.templateMetadata.progressIndicatorStep');
+        // handle cancel
+        if (task === 'cancel' || task === 'backToTemplates') {
+
+            //Set off init here to reset view
+            var mode = component.get('v.templateMetadata.mode');
+            if (mode === 'edit') {
+                model.init(component);
+            }
+            model.getTemplateMetadata().cancel();
+
+        } else if (task === 'next' || task === 'back' || task === 'save') {
+
             var isValid = true;
             var possibleError = '';
+            var step = component.get('v.templateMetadata.progressIndicatorStep');
+
+            // check validity and load values
             if (step === '1') {
                 model.getTemplateInfo().load(component.get('v.templateInfo'));
                 isValid = model.getTemplateInfo().isValid();
                 possibleError = component.get('v.templateMetadata.labels.missingNameDescriptionError');
-                model.getTemplateMetadata().nextStep(isValid, possibleError);
             } else if (step === '2') {
                 //handle template selection and copying here
             } else if (step === '3') {
-                model.getTemplateFields().updateToActive(component.get('v.templateFields').fieldGroups);
+                // model.getTemplateFields().updateToActive(component.get('v.templateFields').fieldGroups);
                 possibleError = model.getTemplateFields().getRequiredFieldErrors();
                 isValid = (possibleError.length === 0);
-                model.getTemplateMetadata().nextStep(isValid, possibleError);
```

Comment:
What does removing this do?

Reply:
It looks like the arguments being passed in here weren't doing anything - https://github.com/SalesforceFoundation/Cumulus/blob/1b7bbc23f9f21a060355324970fe63deaa149d10/src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js#L797

What's that about?